game-core/src/main/java/games/strategy/triplea/ai/pro/ProPurchaseAi.java

```diff
@@ -2339,6 +2338,13 @@ private void upgradeUnitsWithRemainingPUs(
     }
   }
 
+  private static double findUpgradeUnitEfficiency(
+      final ProPurchaseOption ppo, final double strategicValue) {
+    return strategicValue >= 1
+        ? ppo.getAttackEfficiency() * ppo.getDefenseEfficiency() * ppo.getCost() / ppo.getQuantity()
```

Comment:
nit, there is some redundancy in the two conditional branches. Seems can be consolidated with something like:
```
double multiplier = (strategicValue >= 1) ? ppo.getDefenseEfficiency() : ppo.getMovement();
return ppo.getAttackEfficiency() * multiplier * ppo.getCost() / ppo.getQuantity();
```
What do you think?

Reply:
Updated.